monai/transforms/transform.py

```diff
@@ -241,6 +244,21 @@ def __call__(self, data: Any):
         raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
 
 
+class TorchTransform(Transform):
```

Comment:
in the future we may have implementations in `numba`/`cucim`/`cupy`/`jax`, so in my opinion we should use class variable such as `Transform.backends = ["numpy", "torch"]`  to store this information, instead of new classes. `Transform` could have public APIs to query the backend compatibility